theano/gof/op.py

```diff
@@ -783,6 +783,9 @@ def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
         obj = object.__new__(cls)
+        # Ensuring all the Ops created henceforth has __props__
+        if not hasattr(obj, '__props__'):
+            raise
```

Comment:
This one is consistent in raising the error when an Op's instance has `__init__` but not `__props__`. But until we're creating `__props__` for all the Op, it is wouldn't be possible to cache using a Metaclass (the details of which I've mailed you)